geth/common/queued_tx.go

```diff
@@ -0,0 +1,173 @@
+package common
+
+import (
+	"context"
+	"sync"
+
+	"github.com/ethereum/go-ethereum/common"
+)
+
+// QueuedTx holds enough information to complete the queued transaction.
+type QueuedTx struct {
+	id         QueuedTxID
+	hash       common.Hash
+	ctx        context.Context
+	args       SendTxArgs
+	inProgress bool // true if transaction is being sent
+	done       chan struct{}
+	discard    chan struct{}
+	err        error
+	sync.RWMutex
+}
+
+// NewQueuedTx QueuedTx constructor.
+func NewQueuedTx(ctx context.Context, id QueuedTxID, args SendTxArgs) *QueuedTx {
+	return &QueuedTx{
+		id:      id,
+		ctx:     ctx,
+		args:    args,
+		done:    make(chan struct{}, 1),
+		discard: make(chan struct{}, 1),
+	}
+}
+
+// Set updates transaction.
+func (tx *QueuedTx) Set(t *QueuedTx) {
```

Comment:
This is extremely confusing. What is a use case for this? The result is that two `QueuedTx` share the same context and channels.